official/wasm/cmd/cmd.go

```diff
@@ -0,0 +1,127 @@
+package cmd
+
+import (
+	"os"
+	"path/filepath"
+	"sort"
+	"strings"
+
+	"github.com/ignite/cli/v28/ignite/pkg/cliui/colors"
+	"github.com/ignite/cli/v28/ignite/pkg/xgenny"
+	"github.com/ignite/cli/v28/ignite/services/chain"
+	"github.com/spf13/cobra"
+)
+
+// NewWasm creates a new wasm command that holds
+// some other sub commands related to CosmWasm.
+func NewWasm() *cobra.Command {
+	c := &cobra.Command{
+		Use:           "wasm [command]",
+		Aliases:       []string{"w"},
+		Short:         "Ignite wasm integration",
+		SilenceUsage:  true,
+		SilenceErrors: true,
+	}
+
+	// add sub commands.
+	c.AddCommand(
+		NewWasmAdd(),
+		NewWasmConfig(),
+	)
+	return c
+}
+
+const (
+	flagPath = "path"
+	flagHome = "home"
+
+	statusScaffolding  = "Scaffolding..."
+	statusAddingConfig = "Adding config..."
+)
+
+var (
+	modifyPrefix = colors.Modified("modify ")
+	createPrefix = colors.Success("create ")
+	removePrefix = func(s string) string {
+		return strings.TrimPrefix(strings.TrimPrefix(s, modifyPrefix), createPrefix)
+	}
+)
+
+func flagSetPath(cmd *cobra.Command) {
+	cmd.Flags().StringP(flagPath, "p", ".", "path of the app")
+}
+
+func flagSetHome(cmd *cobra.Command) {
+	cmd.Flags().String(flagHome, "", "directory where the blockchain node is initialized")
+}
+
+func getPath(cmd *cobra.Command) string {
+	path, _ := cmd.Flags().GetString(flagPath)
+	return path
+}
+
+func getHome(cmd *cobra.Command) string {
+	home, _ := cmd.Flags().GetString(flagHome)
+	return home
+}
+
+// newChainWithHomeFlags create new *chain.Chain with home and path flags.
+func newChainWithHomeFlags(cmd *cobra.Command, chainOption ...chain.Option) (*chain.Chain, error) {
+	// Check if custom home is provided
+	if home := getHome(cmd); home != "" {
+		chainOption = append(chainOption, chain.HomePath(home))
+	}
+
+	appPath := getPath(cmd)
+	absPath, err := filepath.Abs(appPath)
+	if err != nil {
+		return nil, err
+	}
+
+	return chain.New(absPath, chainOption...)
+}
+
+// sourceModificationToString output the modifications into a readable text.
+func sourceModificationToString(sm xgenny.SourceModification) (string, error) {
```

Comment:
Does it make sense to change Ignite CLI to make this function public? It already exists in Ignite CLI so it could be moved to `xgenny` package and made public?

Reply:
Also, if so then is should be tested as part of `xgenny` which would allow us to remove the `Test_relativePath` defined in `official/wasm/cmd/cmd_test.go`

Reply:
opening an issue in the CLI so we can do it